api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1184,6 +1184,9 @@ message Cluster {
   // If ``connection_pool_per_downstream_connection`` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // TCP protocol options that are applied to TCP connections from pool.
+  core.v3.TcpProtocolOptions tcp_protocol_options = 57;
```

Comment:
sorry I think I wasn't clear on how this works for HTTP :-/  For HTTP, we have this:

https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/upstreams/http/v3/http_protocol_options.proto

the deal is that you can stuff largely any proto in typed_extension_protocol_options  here:
https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/cluster/v3/cluster.proto#L937

so don't need to declare them in the cluster proto, and folks not needing that proto type don't need to pull it in.  So I think we want to treat TCP as we do HTTP - have a standalone proto (which can live in its own file) and it's added via that type extension config field.  Make sense?

/wait

Reply:
@alyssawilk oh thanks for letting me know. It takes some time to make me aware most of the protocol_options field in cluster.proto are deprecated
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto

HTTP proto structure is like
```
api/envoy/extensions/upstreams/http $ tree .
.
├── generic
│   └── v3
│       ├── BUILD
│       └── generic_connection_pool.proto
├── http
│   └── v3
│       ├── BUILD
│       └── http_connection_pool.proto
├── tcp
│   └── v3
│       ├── BUILD
│       └── tcp_connection_pool.proto
└── v3
    ├── BUILD
    └── http_protocol_options.proto
```

current tcp is 
```
api/envoy/extensions/upstreams/tcp $ tree
.
└── generic
    └── v3
        ├── BUILD
        └── generic_connection_pool.proto
```

So i'm going to add a v3 folder at the root of tcp and put the idle timeout into new standalone tcp_protocol_options.proto. the sturcture will be 
```
api/envoy/extensions/upstreams/tcp $ tree
.
├── generic
│   └── v3
│       ├── BUILD
│       └── generic_connection_pool.proto
└── v3
    ├── BUILD
    └── tcp_protocol_options.proto
```

sounds like a plan? @alyssawilk

Reply:
I'm going to let @mattklein123 as I think it's more of an API shephard call :-)
cc @envoyproxy/api-shepherds

Reply:
Yup this sounds good to me. Thank you.

Reply:
Updated. fwiw i hit
` This change appears to add visibility rules. Please get senior maintainer approval to add an exemption to check_visibility tools/code_format/check_format.py` since I make the new proto public visibility